### PR TITLE
enforce string length in byte rather than characters

### PR DESCRIPTION
**Pull request by mikaelarguedas**
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/2567/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/164/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/2037/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/2676/)

related to ros2/ros2#384

### CODE REVIEW

File: rosidl_generator_py/resource/_msg.py.em [hunk 1]
```diff
@@ -211,7 +211,7 @@ bound = 2**nbits
 @[      end if]@
 @[  elif field.type.string_upper_bound]@
             ((isinstance(value, str) or isinstance(value, UserString)) and
-             len(value) <= @(field.type.string_upper_bound))
+             len(value.encode('utf-8')) <= @(field.type.string_upper_bound))
```
Comment (dirk-thomas) [inline rosidl_generator_py/resource/_msg.py.em]: Since we currently don't specify the encoding in the design doc I don't this can assume `utf-8` here.

Comment (mikaelarguedas) [inline rosidl_generator_py/resource/_msg.py.em]: I'm assuming an encoding that uses single byte. I don't think that the actual encoding matters much in this case, does it ? What approach would you suggest here ?

Comment (dirk-thomas) [inline rosidl_generator_py/resource/_msg.py.em]: I am not sure atm. I think the design doc needs to be clear in terms of strings / wstrings before I know how this should be implemented.

**Comment by mikaelarguedas** [comment]
moving this back in `ready` until a design doc specifying strings and char is updated

**Comment by nuclearsandwich** [comment]
The package this pull request targets has been moved to https://github.com/ros2/rosidl_python.git

**Comment by sloretz** [comment]
Closing since package has been moved

**Comment by mikaelarguedas** [comment]
Can you please link to the corresponding PR opened on the new repository ?

(and delete this branch when you're done)

**Comment by sloretz** [comment]
@mikaelarguedas I'm not sure that there needs to be a PR on the new repo. It seems like the result of [this discussion](https://github.com/ros2/rosidl/pull/217#discussion_r114379545) is we don't know how this should be implemented.  Would you like me to delete the branch anyways?

**Comment by mikaelarguedas** [comment]
My recollection from the Bouncy release meeting was that we wanted to get this discussed/solved by Bouncy or shortly after hence the milestone.

If the decision is to not merge the change as is, I think we would be better off with a follow-up ticket for this to be visible/discussed/addressed rather than by closing this PR without follow-up ticket and leaving the branch around without any reference to it.

**Comment by sloretz** [comment]
@mikaelarguedas I'm not sure what would be into a follow-up ticket. If I understand correctly this PR is a possible solution to ros2/ros2#384 "enforce string length in IDL in all languages" which looks like a ticket to implement ros2/design#117. Are you saying this PR is addressing another issue discussed in the bouncy release meeting?

**Comment by mikaelarguedas** [comment]
oh I see your point, as this PR was linked from the higher level ticket I guess that's fine to just close this and delete the branch :+1:

**Pull request closed without merging**